src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -1,227 +1,197 @@
-//                return false;
-//            }
-//
-//            // state check
-//            EditPersonDescriptor e = (EditPersonDescriptor) other;
-//
-//            return getName().equals(e.getName())
-//                    && getPhone().equals(e.getPhone())
-//                    && getEmail().equals(e.getEmail())
-//                    && getAddress().equals(e.getAddress())
-//                    && getTags().equals(e.getTags());
-//        }
-//    }
-//}
+
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_STATUS;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.show.Name;
+import seedu.address.model.show.Show;
+import seedu.address.model.show.Status;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Edits the details of an existing show in Trackermon.
+ */
+public class EditCommand extends Command {
+
+    public static final String COMMAND_WORD = "edit";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the show identified "
+            + "by the index number used in the displayed show list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_NAME + "NAME] "
+            + "[" + PREFIX_STATUS + "NAME] "
+            + "[" + PREFIX_TAG + "TAG]...\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_NAME + "Attack on Titan "
+            + PREFIX_STATUS + "watching";
+
+    public static final String MESSAGE_EDIT_SHOW_SUCCESS = "Edited Show: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_SHOW = "This show already exists in Trackermon.";
+
+    private final Index index;
+    private final EditShowDescriptor editShowDescriptor;
+
+    /**
+     * @param index of the show in the filtered show list to edit
+     * @param editShowDescriptor details to edit the show with
+     */
+    public EditCommand(Index index, EditShowDescriptor editShowDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editShowDescriptor);
+
+        this.index = index;
+        this.editShowDescriptor = new EditShowDescriptor(editShowDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
```

Comment:
Should this function have a JavaDoc comment?

Reply:
cmiiw overrided methods inherit Javadocs from parent